File: src/com/werfad/MarksComponent.java
package com.werfad;

import javax.swing.*;
import java.awt.*;

public class MarksComponent extends JComponent {

    @Override
    public void paint(Graphics g) {
        g.drawString("Hello world", 400, 300);
        super.paint(g);
    }

    public static void main(String[] args) {
        JFrame frame = new JFrame("Hello world");
        frame.setSize(800, 600);
        frame.add(new MarksComponent());
        frame.setVisible(true);
    }

}

File: src/com/werfad/MarksComponent.kt
package com.werfad

import java.awt.Graphics
import javax.swing.JComponent
import javax.swing.JFrame

class MarksComponent : JComponent() {

    override fun paint(g: Graphics) {
        g.drawString("Hello world", 400, 300)
        super.paint(g)
    }

    companion object {

        @JvmStatic
        fun main(args: Array<String>) {
            val frame = JFrame("Hello world")
            frame.setSize(800, 600)
            frame.add(MarksComponent())
            frame.isVisible = true
        }
    }

}
